report/sla.go

```diff
@@ -404,3 +404,38 @@ func SLASummary(probers []probe.Prober) string {
 	summary += "\n" + global.FooterString()
 	return summary
 }
+
+// SLACSVSection set the CSV format for SLA
+func SLACSVSection(r *probe.Result) string {
+	text := "%s, %s, up(%s), down(%s), %.2f%%, %d(%s), %s, %s, %s"
+	return fmt.Sprintf(text, r.Name, r.Endpoint,
+		DurationStr(r.Stat.UpTime), DurationStr(r.Stat.DownTime), r.SLAPercent(),
+		r.Stat.Total, SLAStatusText(r.Stat, Text),
+		r.StartTime.UTC().Format(r.TimeFormat),
+		r.Status.String(), r.Message)
+}
+
+// SLACSV return a full stat report with CSV format
+func SLACSV(probers []probe.Prober) string {
+	csv := "Name, Endpoint, UpTime, DownTime, SLA, ProbeSummary, LatestProbe, LatestStatus, Message\n"
+	for _, p := range probers {
+		csv += SLACSVSection(p.Result()) + "\n"
```

Comment:
propose to use `bytes.Buffer` to improve performance in case there are many probers.

Reply:
and propose to use `encoding/csv` to generate the CSV result, it handles corner cases better.